openmeter/ent/schema/billing.go

```diff
@@ -0,0 +1,358 @@
+package schema
+
+import (
+	"database/sql"
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"entgo.io/ent"
+	"entgo.io/ent/schema/edge"
+	"entgo.io/ent/schema/field"
+	"entgo.io/ent/schema/index"
+	"github.com/alpacahq/alpacadecimal"
+
+	"github.com/openmeterio/openmeter/openmeter/billing"
+	"github.com/openmeterio/openmeter/openmeter/billing/invoice"
+	"github.com/openmeterio/openmeter/openmeter/billing/provider"
+	"github.com/openmeterio/openmeter/pkg/framework/entutils"
+)
+
+type BillingProfile struct {
+	ent.Schema
+}
+
+func (BillingProfile) Mixin() []ent.Mixin {
+	return []ent.Mixin{
+		entutils.IDMixin{},
+		entutils.NamespaceMixin{},
+		entutils.TimeMixin{},
+	}
+}
+
+func (BillingProfile) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("key").
+			NotEmpty().
+			Immutable(),
+		field.String("provider_config").
+			GoType(provider.Configuration{}).
+			ValueScanner(ProviderConfigValueScanner).
+			SchemaType(map[string]string{
+				"postgres": "jsonb",
+			}),
+		field.JSON("billing_config", billing.Configuration{}).
+			SchemaType(map[string]string{
+				"postgres": "jsonb",
+			}),
+		field.Bool("default").
+			Default(false),
+	}
+}
+
+func (BillingProfile) Edges() []ent.Edge {
+	return []ent.Edge{
+		edge.To("billing_invoices", BillingInvoice.Type),
```

Comment:
Please set DB field name. Ent default is weird.

Reply:
That is not required, as the reference is to the id field. At the from side I have set the filed properly. If you check the SQL migrations you will see that there are no weird field names present.